dotnet/src/SemanticKernel.UnitTests/AI/FunctionChoiceBehaviors/FunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,255 @@
+    }
+
+    [Fact]
+    public void AutoFunctionChoiceShouldAdvertiseProvidedFunctions()
+    {
+        // Arrange
+        var plugin = GetTestPlugin();
+        this._kernel.Plugins.Add(plugin);
+
+        // Act
+        var choiceBehavior = FunctionChoiceBehavior.Auto(functions: [plugin.ElementAt(0), plugin.ElementAt(1)]);
+
+        var config = choiceBehavior.GetConfiguration(new() { Kernel = this._kernel });
+
+        // Assert
+        Assert.NotNull(config);
+
+        Assert.NotNull(config.Functions);
+        Assert.Equal(2, config.Functions.Count);
+        Assert.Contains(config.Functions, f => f.Name == "Function1");
+        Assert.Contains(config.Functions, f => f.Name == "Function2");
+    }
+
+    [Fact]
+    public void AutoFunctionChoiceShouldAllowAutoInvocation()
+    {
+        // Arrange
+        var plugin = GetTestPlugin();
+        this._kernel.Plugins.Add(plugin);
+
+        // Act
+        var choiceBehavior = FunctionChoiceBehavior.Auto(autoInvoke: true);
+
+        var config = choiceBehavior.GetConfiguration(new() { Kernel = this._kernel });
+
+        // Assert
+        Assert.NotNull(config);
+        Assert.True(config.AutoInvoke);
+    }
+
+    [Fact]
+    public void AutoFunctionChoiceShouldAllowManualInvocation()
+    {
+        // Arrange
+        var plugin = GetTestPlugin();
+        this._kernel.Plugins.Add(plugin);
+
+        // Act
+        var choiceBehavior = FunctionChoiceBehavior.Auto(autoInvoke: false);
+
+        var config = choiceBehavior.GetConfiguration(new() { Kernel = this._kernel });
+
+        // Assert
+        Assert.NotNull(config);
+        Assert.False(config.AutoInvoke);
+    }
+
+    //[Fact]
+    //public void RequiredFunctionChoiceShouldAdvertiseKernelFunctions()
+    //{
+    //    // Arrange
+    //    var plugin = GetTestPlugin();
+    //    this._kernel.Plugins.Add(plugin);
+
+    //    // Act
+    //    var choiceBehavior = FunctionChoiceBehavior.Required(functions: null);
+
+    //    var config = choiceBehavior.GetConfiguration(new() { Kernel = this._kernel });
+
+    //    // Assert
+    //    Assert.NotNull(config);
+
+    //    Assert.NotNull(config.Functions);
+    //    Assert.Equal(3, config.Functions.Count);
+    //    Assert.Contains(config.Functions, f => f.Name == "Function1");
+    //    Assert.Contains(config.Functions, f => f.Name == "Function2");
+    //    Assert.Contains(config.Functions, f => f.Name == "Function3");
+    //}
+
+    //[Fact]
+    //public void RequiredFunctionChoiceShouldAdvertiseProvidedFunctions()
+    //{
+    //    // Arrange
+    //    var plugin = GetTestPlugin();
+    //    this._kernel.Plugins.Add(plugin);
+
+    //    // Act
+    //    var choiceBehavior = FunctionChoiceBehavior.Required(functions: [plugin.ElementAt(0), plugin.ElementAt(1)]);
+
+    //    var config = choiceBehavior.GetConfiguration(new() { Kernel = this._kernel });
+
+    //    // Assert
+    //    Assert.NotNull(config);
```

Comment:
Many commented out...

Reply:
They will be needed later to test other parts of the new function calling model when they are added one after another. It'll be challenging to keep track of tests that were copied from the old big function calling abstraction PR and which were not. So, I'll keep them commented out for a while.